pydantic/functional_serializers.py

```diff
@@ -39,8 +40,13 @@ def __get_pydantic_core_schema__(
             The Pydantic core schema.
         """
         schema = handler(source_type)
-        return_type = _decorators.get_function_return_type(self.func, self.return_type)
-        return_schema = None if return_type is None else handler.generate_schema(return_type)
+        try:
+            return_type = _decorators.get_function_return_type(
+                self.func, self.return_type, handler.get_types_namespace()
+            )
```

Comment:
Hmm if this is the only place this is being used I'm inclined to make `_get_types_namespace` a private attribute and just `type: ignore` it, only exposing it when someone else asks for it.